src/realm/sync/noinst/protocol_codec.cpp

```diff
@@ -215,13 +215,16 @@ void ServerProtocol::insert_single_changeset_download_message(OutputBuffer& out,
     entry.changeset.write_to(out);
 
     if (logger.would_log(util::Logger::Level::trace)) {
+        util::AppendBuffer<char> changeset_buffer;
+        entry.changeset.copy_to(changeset_buffer);
```

Comment:
The compaction code removed in RCORE-2205 did a bunch of decoding/re-encoding of changesets before they got to be logged here, so `entry.changeset.get_first_chunk()` was only valid if the `ChunkedBinaryData` in `changeset` wrapped a single `BinaryData`. We just copy the whole (potentially chunked) changeset into a buffer here to simulate the old behavior.